### ISSUE TITLE
Support Java auto-instrumentation image 2.x

### ISSUE THREAD

**Issue by pavolloffay** [open-telemetry/opentelemetry-operator#2529]
### Is your feature request related to a problem? Please describe.

Crated from https://github.com/open-telemetry/opentelemetry-operator/pull/2514#issuecomment-1894092595

The Java agent 2.x has several breaking changes https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v2.0.0:
* exporter changed to OTLP HTTP
* instrumentation changes 

### Describe the solution you'd like

Operator should configure HTTP exporter if 2.x java agent is used.

**Comment by pavolloffay** [comment]
The image gets published when https://github.com/open-telemetry/opentelemetry-operator/blob/main/autoinstrumentation/java/version.txt#L1 changes. It should be fine to use it for both version streams.

**Comment by TylerHelmuth** [comment]
@pavolloffay you're right. I coded up a solution that let's us maintain 2.x and 1.x versions simultaneously, but if you're ok with the value in `versions.txt` moving back and forth between `1.x` and `2.x` that is simpler.

**Comment by msvechla** [comment]
how can we use v2 of the java instrumentation with the current operator? Is this supported?

**Comment by atoulme** [comment]
Checking https://github.com/open-telemetry/opentelemetry-operator/blob/main/autoinstrumentation/java/version.txt we are now using 2.x. Closing as complete.
